docs/csharp/asynchronous-programming/async-scenarios.md

```diff
@@ -1,97 +1,91 @@
+In the next example, a mobile game inflicts damage on several agents on the screen in response to a button event. Performing the damage calculation can be expensive. Running the calculation on the UI thread can cause display and UI interaction issues during the calculation.
 
-Here are two questions you should ask before you write any code:
+The best way to handle the task is to start a background thread to complete the work with the `Task.Run` method. The operation waits on the result by using an `await` expression. This approach allows the UI to run smoothly while the work completes in the background.
 
-1. Will your code be "waiting" for something, such as data from a database?
+:::code language="csharp" source="snippets/async-scenarios/Program.cs" ID="PerformGameCalculation":::
 
-   If your answer is "yes", then your work is **I/O-bound**.
+The code clearly expresses the intent of the button `Clicked` event. It doesn't require managing a background thread manually, and it completes the task in a nonblocking manner.
 
-1. Will your code be performing an expensive computation?
+## Recognize CPU-bound and I/O-bound scenarios
 
-   If you answered "yes", then your work is **CPU-bound**.
+The previous examples demonstrated how to use the `async` and `await` expressions for I/O-bound and CPU-bound work. An example was provided for each scenario to showcase how the code is different based on where the operation is bound. To prepare for your implementation, you need to understand how to identify when an operation is I/O-bound or CPU-bound. Your implementation choice can greatly affect the performance of your code and potentially lead to misusing constructs.
 
-If the work you have is **I/O-bound**, use `async` and `await` *without* `Task.Run`. You *should not* use the Task Parallel Library.
+There are two primary questions to address before you write any code:
 
-If the work you have is **CPU-bound** and you care about responsiveness, use `async` and `await`, but spawn off the work on another thread *with* `Task.Run`. If the work is appropriate for concurrency and parallelism, also consider using the [Task Parallel Library](../../standard/parallel-programming/task-parallel-library-tpl.md).
+| Question | Scenario | Implementation |
+| --- | --- | --- |
+| _Should the code wait for a result or action, such as data from a database?_ | **I/O-bound** | Use `async` and `await` expressions _without_ the `Task.Run` method. <br><br> Avoid using the Task Parallel Library. |
+| _Should the code run an expensive computation?_ | **CPU-bound** | Use `async` and `await` expressions, but spawn off the work on another thread with the `Task.Run` method. This approach addresses concerns with CPU responsiveness. <br><br> If the work is appropriate for concurrency and parallelism, also consider using the [Task Parallel Library](../../standard/parallel-programming/task-parallel-library-tpl.md). |
```

Comment:
Personally, I think the original reads better. In part, because it's possible that operations are a mix of CPU bound and I/O bound.